lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -342,29 +366,28 @@ async sub transform_transaction {
             timestamp    => $int_timestamp,
         );
 
-        # if the input is not 0x we check the transaction searching by any
-        # transfer event to any contract, this can return more than 1 transaction.
-        # we need to do this before set the transaction type since the `to` address
-        # will change in case it be a contract.
-        # the node response for an empty input is 0x
-        if ($decoded_transaction->{input} ne '0x') {
-            $transaction = await $self->_check_contract_transaction($transaction) if $transaction;
+        my @transactions = await $self->_check_contract_transaction($transaction, $receipt);
+
+        unless (scalar @transactions) {
```

Comment:
I'd move this into the _check_contract_transaction so that you are always guaranteed an array from it.

Reply:
moving this to `_check_contract_transaction` will return a transaction even if it's not a contract, so I think it's better to be outside, also it is better to add the unit tests for the contracts going throw this method.